price-discovery/src/lib.rs

```diff
@@ -243,4 +290,27 @@ pub trait PriceDiscovery:
             "Unbond period not finished yet"
         );
     }
+
+    fn require_launched_token_over_min_price(&self) {
+        let launched_token_id = self.launched_token_id().get();
+        let accepted_token_id = self.accepted_token_id().get();
+
+        let min_price = self.min_launched_token_price().get();
+        let launched_token_balance = self.blockchain().get_sc_balance(&launched_token_id, 0);
+        let accepted_token_balance = self.blockchain().get_sc_balance(&accepted_token_id, 0);
+
+        if accepted_token_balance == 0 {
+            return;
+        }
+
+        require!(launched_token_balance > 0, "No launched tokens available");
+        require!(
+            accepted_token_balance * MIN_PRICE_PRECISION / launched_token_balance >= min_price,
```

Comment:
maybe extract this price compute in a variable

Reply:
I think `MIN_PRICE_PRECISION` should be set based on `launched_token` decimals numbers

Reply:
This can be changed on deploy - I do not see it necessary to be in config or kept in storage